tests/phpunit/tests/block-templates-registry.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Tests for the Block Templates Registry.
+ *
+ * @package WordPress
+ * @subpackage Block Templates
+ *
+ * @group block-templates
+ */
+class Tests_Block_Templates_Registry extends WP_UnitTestCase {
+	/**
+	 * @var WP_Block_Templates_Registry
+	 */
+	private $registry;
+
+	public function set_up() {
+		parent::set_up();
+		$this->registry = new WP_Block_Templates_Registry();
+	}
+
+	/**
+	 * Data provider for test_template_name_validation.
+	 *
+	 * @return array[] Test data.
+	 */
+	public function data_template_name_validation() {
```

Comment:
Data provider methods should be static:
```suggestion
	public static function data_template_name_validation() {
```